src/Libraries/Microsoft.Extensions.AI/Functions/AIFunctionFactoryOptions.cs

```diff
@@ -55,4 +56,91 @@ public AIFunctionFactoryOptions()
+    /// will be invoked for that parameter each time the <see cref="AIFunction"/> is invoked. The <see cref="BindParameter"/>
+    /// delegate will be passed the <see cref="ParameterInfo" /> for the parameter, the <see cref="AIFunctionArguments"/> used
+    /// with the <see cref="AIFunction.InvokeAsync"/> call, and the <see cref="ParameterBindingOptions"/> result produced by
+    /// the <see cref="ConfigureParameterBinding"/> delegate. The return value of the delegate will be used for the parameter's
+    /// value. If <see cref="ParameterBindingOptions"/> is <see langword="null"/> or if the <see cref="ParameterBindingOptions"/>
+    /// it produces has <see cref="ParameterBindingOptions.UseBindParameter"/> as <see langword="false"/>, this delegate will be
+    /// ignored for that parameter.
+    /// </remarks>
+    public Func<ParameterBindingOptions, ParameterInfo, AIFunctionArguments, object?>? BindParameter { get; set; }
+
+    /// <summary>Provides configuration options produced by the <see cref="ConfigureParameterBinding"/> delegate.</summary>
+    [StructLayout(LayoutKind.Auto)]
+    public readonly record struct ParameterBindingOptions
+    {
+        /// <summary>Gets a value indicating whether the <see cref="BindParameter"/> delegate should be used for this parameter.</summary>
+        /// <remarks>
+        /// <para>
+        /// The default value is <see langword="false"/>.
+        /// </para>
+        /// <para>
+        /// If this property is <see langword="false"/>, the associated parameter will employ the default binding.
+        /// If this property is <see langword="true"/>, the <see cref="BindParameter"/> delegate will be invoked to be
+        /// solely responsible for the value to use for the property.
+        /// </para>
+        /// <para>
+        /// Typically, this value is used in combination with <see cref="ExcludeFromSchema"/> also set to <see langword="true"/>,
+        /// for cases where <see cref="BindParameter"/> will source the argument value for the parameter from somewhere other than
+        /// the argument dictionary (for example, from its <see cref="AIFunctionArguments.Services"/> or
+        /// <see cref="AIFunctionArguments.Context"/>. However, it may be used even when <see cref="ExcludeFromSchema"/> is <see langword="false"/>,
+        /// in a situation where it's desirable for the AI service to generate a value for the parameter, which <see cref="BindParameter"/>
+        /// may then use as part of its logic.
+        /// </para>
+        /// <para>
+        /// It is an error for <see cref="ConfigureParameterBinding"/> to return a <see cref="ParameterBindingOptions"/> with
+        /// <see cref="UseBindParameter"/> set to <see langword="true"/> when <see cref="BindParameter"/> is <see langword="null"/>.
+        /// </para>
+        /// </remarks>
+        public bool UseBindParameter { get; init; }
```

Comment:
Why not just specify an individual binding delegate for each parameter binding options? I'm assuming you could avoid allocating a new delegate for every parameter being examined.

Reply:
Hmm, they would probably simplify a few things. Will try.

Reply:
Done.